tests/ha/test_server_pod_failure.py

```diff
@@ -806,5 +806,391 @@ def test_degraded_deletes_safe_server_pod_shutdown(self):
+        del_total_bkt = HA_CFG["s3_bucket_data"]["no_buckets_for_deg_deletes"]
+        wr_bucket = HA_CFG["s3_bucket_data"]["no_bck_writes"]
+        del_bucket = HA_CFG["bg_bucket_ops"]["no_del_buckets"]
+        wr_output = Queue()
+        del_output = Queue()
+
+        LOGGER.info("Creating IAM user.")
+        users = self.mgnt_ops.create_account_users(nusers=1)
+        access_key = list(users.values())[0]["accesskey"]
+        secret_key = list(users.values())[0]["secretkey"]
+        self.test_prefix = 'test-39912'
+        self.s3_clean = users
+        s3_test_obj = S3TestLib(access_key=access_key, secret_key=secret_key,
+                                endpoint_url=S3_CFG["s3_url"])
+
+        LOGGER.info("Step 1: Start continuous WRITEs and DELETEs with variable object sizes "
+                    "during server pod down by delete deployment.")
+
+        LOGGER.info("Perform WRITEs on %s buckets for background DELETEs", del_total_bkt)
+        args = {'test_prefix': self.test_prefix, 'test_dir_path': self.test_dir_path,
+                'skipget': True, 'skipdel': True, 'bkts_to_wr': del_total_bkt, 'output': wr_output}
+        self.ha_obj.put_get_delete(event, s3_test_obj, **args)
+        wr_resp = ()
+        while len(wr_resp) != 3:
+            wr_resp = wr_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        s3_data = wr_resp[0]  # Contains IAM user data for passed buckets
+        buckets = s3_test_obj.bucket_list()[1]
+        assert_utils.assert_equal(len(buckets), del_total_bkt,
+                                  f"Failed to create {del_total_bkt} number of buckets."
+                                  f"Created {len(buckets)} number of buckets")
+        LOGGER.info("Performed WRITEs on %s buckets for background DELETEs", del_total_bkt)
+
+        LOGGER.info("Starting WRITEs on %s buckets in background", wr_bucket)
+        args = {'test_prefix': self.test_prefix, 'test_dir_path': self.test_dir_path,
+                'skipget': True, 'skipdel': True, 'bkts_to_wr': wr_bucket, 'output': wr_output}
+        thread1 = threading.Thread(target=self.ha_obj.put_get_delete,
+                                   args=(event, s3_test_obj,), kwargs=args)
+
+        LOGGER.info("Starting DELETEs of %s buckets in background", del_bucket)
+        # Get the list of buckets name from already written buckets
+        written_bck = list(s3_data.keys())
+        get_random_buck = random.sample(written_bck, del_bucket)
+        args = {'test_prefix': self.test_prefix, 'test_dir_path': self.test_dir_path,
+                'skipput': True, 'skipget': True, 'bkt_list': get_random_buck,
+                'bkts_to_del': del_bucket, 'output': del_output}
+        thread2 = threading.Thread(target=self.ha_obj.put_get_delete,
+                                   args=(event, s3_test_obj,), kwargs=args)
+        thread1.daemon = thread2.daemon = True  # Daemonize thread
```

Comment:
Are you sure that this won't act as assignment operator?
Otherwise thread1.daemon will be equal to thread2.daemon.

Reply:
> Are you sure that this won't act as assignment operator? Otherwise thread1.daemon will be equal to thread2.daemon.

Yes it won't act as assignment operator.